packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -6153,6 +6129,112 @@ addEventListener('fetch', event => {});`
 				expect(std.warn).toMatchInlineSnapshot(`""`);
 			});
+
+			it("should error for compiled logfwdr schemas", async () => {
+				writeWranglerToml({
+					logfwdr: {
+						capnp_schema: "./message.capnp.compiled",
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+
+				await expect(() => runWrangler("deploy index.js")).rejects
+					.toThrowErrorMatchingInlineSnapshot(`
+			"Processing wrangler.toml configuration:
+			  - \\"logfwdr\\" bindings \\"capnp_schema\\" field should not be pre-compiled - Wrangler will run capnp for you."
+		`);
+			});
+
+			it("should when a logfwdr schema isn't specified", async () => {
+				writeWranglerToml({
+					logfwdr: {
+						capnp_schema: undefined,
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+
+				await expect(() => runWrangler("deploy index.js")).rejects
+					.toThrowErrorMatchingInlineSnapshot(`
+			"Processing wrangler.toml configuration:
+			  - \\"logfwdr\\" bindings should have a string \\"capnp_schema\\" field but got {\\"bindings\\":[{\\"name\\":\\"httplogs\\",\\"destination\\":\\"httplogs\\"},{\\"name\\":\\"trace\\",\\"destination\\":\\"trace\\"}]}."
+		`);
+			});
+
+			it("should error when the capnp CLI tool is not present", async () => {
+				(callCapnp as jest.MockedFunction<typeof callCapnp>).mockImplementation(
+					() => {
+						throw new Error(
+							"The capnp compiler is required to upload capnp schemas, but is not present."
+						);
+					}
+				);
+
+				writeWranglerToml({
+					logfwdr: {
+						capnp_schema: "./message.capnp",
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+				writeWorkerSource();
+
+				await expect(() =>
+					runWrangler("deploy index.js")
+				).rejects.toThrowErrorMatchingInlineSnapshot(
+					`"The capnp compiler is required to upload capnp schemas, but is not present."`
+				);
+			});
+
+			it("should error when the old schema property is used", async () => {
+				writeWranglerToml({
+					logfwdr: {
+						// @ts-expect-error This should be erroring
+						schema: "./message.capnp",
+						bindings: [
+							{
+								name: "httplogs",
+								destination: "httplogs",
+							},
+							{
+								name: "trace",
+								destination: "trace",
+							},
+						],
+					},
+				});
+
+				await expect(() => runWrangler("deploy index.js")).rejects
+					.toThrowErrorMatchingInlineSnapshot(`
+			"Processing wrangler.toml configuration:
+			  - \\"logfwdr\\" binding \\"schema\\" property has been replaced with \\"capnp_schema\\", and no longer requires you to manually call capnp."
```

Comment:
I would be more explicit about what is expected in the new field. E.g.

```suggestion
			  - \\"logfwdr\\" binding \\"schema\\" property has been replaced with \\"capnp_schema\\", which expects a file path to an uncompiled capnp schema."
```